source/Date.cpp

```diff
@@ -1,23 +1,72 @@
 
 #include "Date.h"
-#include <chrono>
 
-Date::Date(const int& DAY, const int& MONTH, const int& YEAR):
-    YEAR(Y),
-    MONTH(M),
-    DAY(D)
-{}
-now(){
-    auto timenow =
-         chrono::system_clock::to_time_t(chrono::system_clock::now());
-     
-       cout << ctime(&timenow) << endl;
+#include <ctime>
+#include <stdexcept>
+
+Date::Date(uint8_t D, uint8_t M, uint8_t Y) {
+    
+    if(Y < 1970 || Y > 2039 || M > 12 || D > 31) {
+        throw std::logic_error("Invalid date range");
+    }
+    
+    DAY = D;
+    MONTH = M;
+    YEAR = Y;
 }
 
+
 /* Getters*/
-Date getDate() const {
-    
+uint8_t Date::getDAY() const {
+    return DAY;
+}
+
+uint8_t Date::getMONTH() const {
+    return MONTH;
+}
+
+uint8_t Date::getYEAR() const {
+    return YEAR;
 }
 
 /* Setters*/
-void setDate(const int& DAY, const int& MONTH, const int& YEAR);
+void Date::setDate(uint8_t D, uint8_t M, uint8_t Y)
+{
+    if(Y < 1970 || M > 12 || D > 31) {
+        throw std::logic_error("Invalid date range");
+    }
+    
+    DAY = D;
+    MONTH = M;
+    YEAR = Y;
+    
+}
+
+/* Operators */
+Date& Date::operator+(const Date& other) {
+    
+    if(DAY + other.DAY > 12) {
+        MONTH = 1 + (MONTH + 1) % 13;
+    }
+    
+    DAY = 1 + (DAY + other.DAY) % 32;
+    
+    if(MONTH )
+    
+    YEAR += other.YEAR;
+    
+    return *this;
+}
+
+Date& Date::operator-(const Date& other) {
+    
```

Comment:
If not implemented, should be noted in a comment.

Reply:
Done.